projects/manage-supervision-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/service/SentenceService.kt

```diff
@@ -116,6 +119,22 @@ class SentenceService(
         return null
     }
 
+    fun getUnpaidWorkTime(requirementType: String, eventId: Long, eventNumber: String): String? {
+        if (requirementType.equals("W", true)) {
+            val durationInMinutes: Long = upwAppointmentRepository.calculateUnpaidTimeWorked(eventId, eventNumber)
+            durationInMinutes.let {
+                val durationInHours = it / 60
```

Comment:
do we need to think about rounding here?

Reply:
will change the logic for this one

Reply:
updated logic